File: extract_pdfs.js
const fs = require('fs');
const pdf = require('pdf-parse');
const path = require('path');

const pdfDir = 'C:/Users/Life/.gemini/antigravity/MedFit App/forms';
const outputFile = 'C:/Users/Life/.gemini/antigravity/MedFit App/medfit-portal/pdf_texts.txt';

async function extract() {
    const files = fs.readdirSync(pdfDir).filter(f => f.endsWith('.pdf'));
    let result = '';

    for (const file of files) {
        console.log(`Processing ${file}`);
        const dataBuffer = fs.readFileSync(path.join(pdfDir, file));
        try {
            const data = await pdf(dataBuffer);
            result += `--- ${file} ---\n${data.text}\n\n`;
        } catch (e) {
            console.error(`Error with ${file}:`, e.message);
            result += `--- ${file} ---\nError extracting text\n\n`;
        }
    }

    fs.writeFileSync(outputFile, result);
    console.log('Extraction complete. Output saved to ' + outputFile);
}

extract();

File: extract_pdfs.py
try:
    import PyPDF2
except ImportError:
    import os
    os.system("pip install PyPDF2")
    import PyPDF2

import os

pdf_dir = r"C:\Users\Life\.gemini\antigravity\MedFit App\forms"
output_file = r"C:\Users\Life\.gemini\antigravity\MedFit App\medfit-portal\pdf_texts.txt"

with open(output_file, "w", encoding="utf-8") as out:
    for filename in os.listdir(pdf_dir):
        if filename.endswith(".pdf"):
            out.write(f"--- {filename} ---\n")
            filepath = os.path.join(pdf_dir, filename)
            try:
                reader = PyPDF2.PdfReader(filepath)
                text = ""
                for page in reader.pages:
                    text += page.extract_text() + "\n"
                out.write(text)
                out.write("\n\n")
            except Exception as e:
                out.write(f"Error reading {filename}: {e}\n\n")

print("Extraction complete.")
